Network/HTTP/Req.hs

```diff
@@ -937,8 +960,49 @@ uriHost uri = case URI.uriAuthority uri of
   Right URI.Authority {..} ->
     Just (URI.unRText authHost)
 
+-- | A quasiquoter to build a 'Url'. Can be used both as an expression and as a
+-- pattern. Only works for URIs that don't involve query options (e.g. port)
+--
+-- >>> [urlQ|https://exmaple.org/test|]
+-- Url Https ("test" :| ["example.org"])
+urlQ :: TH.QuasiQuoter
+urlQ = TH.QuasiQuoter
+  { quoteExp  = uriQ
+  , quotePat  = \str -> TH.appE [|(==)|] (uriQ str) `TH.viewP` [p|True|]
+  , quoteType = error "This usage is not supported"
+  , quoteDec  = error "This usage is not supported" }
+  where
+    uriQ :: String -> TH.Q TH.Exp
+    uriQ str = do
+      (url, uri) <- urlQShared str
+      case uriOptions uri of
+        Just _ -> fail "URL with options not supported, please use urlWithOptsQ"
+        Nothing -> url
+
+-- | A quasiquoter to build a 'Url' and 'Option' tuple. Can only be used as an
+-- expression.
+--
+-- >>> :t [urlWithOptsQ|https://example.org:443/test?x=1|]
```

Comment:
The same here.

Reply:
I think the examples show, what is meant by options. Also they are useful if you don't know what a quasiquoter is.

Reply:
> what is meant by options

But how? It just repeats the expression verbatim here plus it shows the type. You could as well add the type in the doc string itself.

Reply:
By the difference between the two examples:

```
[urlQ|https://exmaple.org/test|]
[urlWithOptsQ|https://example.org:443/test?x=1|]
```